python-frontend/src/main/java/org/sonar/python/api/PythonPunctuator.java

```diff
@@ -83,7 +83,9 @@ public enum PythonPunctuator implements TokenType {
   LEFT_ASSIGN("<<="),
   MUL_MUL_ASSIGN("**="),
   MATRIX_MULT_ASSIGN("@="),
-  WALRUS_OPERATOR(":=");
+  WALRUS_OPERATOR(":="),
+  QUESTION_MARK("?")
```

Comment:
I think this will make us lex `?` as a valid Python punctuator even outside of IPython context.
Can we avoid that?
If we can't avoid it easily, I'd suggest separating this from the other punctuators and to add a comment that this is used for IPython dynamic object information.

Reply:
done